### DISCUSSION TITLE
Customize `generate_random_paths()`

### DISCUSSION THREAD

**Discussion by HJA24** [networkx/networkx#7642] [Q&A]
Hi,

I want to customize the `generate_random_paths()`-function. In my case the starting node should always be the same. I modify the function slightly as follows:

```
import numpy as np
import networkx as nx
from networkx.utils import np_random_state

@np_random_state(5)
@nx._dispatchable(edge_attrs="weight")
def generate_random_paths(G, sample_size, path_length=5, index_map=None, weight="weight"):
    adj_mat = nx.to_numpy_array(G, weight=weight)
    inv_row_sums = np.reciprocal(adj_mat.sum(axis=1)).reshape(-1, 1)
    transition_probabilities = adj_mat * inv_row_sums

    node_map = list(G)
    num_nodes = G.number_of_nodes()

    for path_index in range(sample_size):
        node_index = 0                                                     # modified this line
        node = node_map[node_index]                           
        path = [node]

        if index_map is not None:
            if node in index_map:
                index_map[node].add(path_index)
            else:
                index_map[node] = {path_index}

        starting_index = node_index

        for _ in range(path_length):
            nbr_index = seed.choice(
                num_nodes, p=transition_probabilities[starting_index]
            )

            starting_index = nbr_index
            nbr_node = node_map[nbr_index]
            path.append(nbr_node)

            if index_map is not None:
                if nbr_node in index_map:
                    index_map[nbr_node].add(path_index)

                else:
                    index_map[nbr_node] = {path_index}

        yield path

G = nx.star_graph(3)
random_path = generate_random_paths(G, 2)
print(random_path)
```

However, this results in an error:

> AttributeError: module 'networkx' has no attribute '_dispatchable'

How do I import this decorator and make the function work as expected?

**Comment by dschult** [comment]
Thanks for these posts. If you have figured out how to solve issue #7641 can you close that issue?

You shouldn't need to use `_dispatchable` for a function in your own work. That is for functions in the package to let backends support that functionality through dispatching to the backend. If you really want to use it try `@nx.utils._dispatchable`. 

Looks like we should update the _dispatchable doc_string examples to include the standard way to import/call the function.
We should also put info in the first paragraph that tells developers/hackers that they don't need to use the decorator when deevloping their own code.
